lib/ecto/schema.ex

```diff
@@ -319,6 +320,16 @@ defmodule Ecto.Schema do
   end
 
   @doc """
+  TODO
+  """
+  defmacro schema(scope, source, [do: block]) do
```

Comment:
I also thought about exposing the `@association_scope` attribute directly instead of defining `schema/3`.

I personally think that this would be better since I'm not scoping custom field types.